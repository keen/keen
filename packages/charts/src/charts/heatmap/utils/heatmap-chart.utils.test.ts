import {
  generateHorizontalBlocks,
  generateVerticalBlocks,
} from './heatmap-chart.utils';

import { heatmapChart } from '../heatmap-chart.fixtures';

describe('<Heatmap /> - utils', () => {
  const data = [
    { label: 'Windows', sale: -3, buy: 11, revenue: 30 },
    { label: 'Android', sale: 12, buy: 3, revenue: 21 },
    { label: 'Linux', sale: -3, buy: 11, revenue: 30 },
  ];

  describe('generateVerticalBlocks()', () => {
    const chart: any = {
      data,
      ...heatmapChart,
    };

    it('should create proper domain for xScale', () => {
      const { xScale } = generateVerticalBlocks(chart);

      expect(xScale.domain()).toMatchInlineSnapshot(`
        Array [
          "sale",
          "buy",
          "revenue",
        ]
      `);
    });

    it('should create proper domain for yScale', () => {
      const { yScale } = generateVerticalBlocks(chart);

      expect(yScale.domain()).toMatchInlineSnapshot(`
        Array [
          "Windows",
          "Android",
          "Linux",
        ]
      `);
    });

    it('should calculate blocks and apply colors', () => {
      const { blocks } = generateVerticalBlocks(chart);

      expect(blocks).toMatchSnapshot();
    });
  });

  describe('generateHorizontalBlocks()', () => {
    const chart: any = {
      data,
      ...heatmapChart,
    };

    it('should create proper domain for xScale', () => {
      const { xScale } = generateHorizontalBlocks(chart);

      expect(xScale.domain()).toMatchInlineSnapshot(`
        Array [
          "Windows",
          "Android",
          "Linux",
        ]
      `);
    });

    it('should create proper domain for yScale', () => {
      const { yScale } = generateHorizontalBlocks(chart);

      expect(yScale.domain()).toMatchInlineSnapshot(`
        Array [
          "sale",
          "buy",
          "revenue",
        ]
      `);
    });

    it('should calculate blocks and apply colors', () => {
      const { blocks } = generateHorizontalBlocks(chart);

      expect(blocks).toMatchSnapshot();
    });
  });
});
