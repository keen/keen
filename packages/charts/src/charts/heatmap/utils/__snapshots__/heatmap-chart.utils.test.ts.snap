// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heatmap /> - utils generateHorizontalBlocks() should calculate blocks and apply colors 1`] = `
Array [
  Object {
    "color": "rgb(255, 0, 0)",
    "height": 123.33333333333333,
    "key": "0.sale.block",
    "selector": Array [
      0,
      "sale",
    ],
    "value": -3,
    "width": 160,
    "x": 10,
    "y": 256.66666666666663,
  },
  Object {
    "color": "rgb(153, 153, 255)",
    "height": 123.33333333333333,
    "key": "1.sale.block",
    "selector": Array [
      1,
      "sale",
    ],
    "value": 12,
    "width": 160,
    "x": 170,
    "y": 256.66666666666663,
  },
  Object {
    "color": "rgb(255, 0, 0)",
    "height": 123.33333333333333,
    "key": "2.sale.block",
    "selector": Array [
      2,
      "sale",
    ],
    "value": -3,
    "width": 160,
    "x": 330,
    "y": 256.66666666666663,
  },
  Object {
    "color": "rgb(162, 162, 255)",
    "height": 123.33333333333333,
    "key": "0.buy.block",
    "selector": Array [
      0,
      "buy",
    ],
    "value": 11,
    "width": 160,
    "x": 10,
    "y": 133.33333333333331,
  },
  Object {
    "color": "rgb(230, 230, 255)",
    "height": 123.33333333333333,
    "key": "1.buy.block",
    "selector": Array [
      1,
      "buy",
    ],
    "value": 3,
    "width": 160,
    "x": 170,
    "y": 133.33333333333331,
  },
  Object {
    "color": "rgb(162, 162, 255)",
    "height": 123.33333333333333,
    "key": "2.buy.block",
    "selector": Array [
      2,
      "buy",
    ],
    "value": 11,
    "width": 160,
    "x": 330,
    "y": 133.33333333333331,
  },
  Object {
    "color": "rgb(0, 0, 255)",
    "height": 123.33333333333333,
    "key": "0.revenue.block",
    "selector": Array [
      0,
      "revenue",
    ],
    "value": 30,
    "width": 160,
    "x": 10,
    "y": 10,
  },
  Object {
    "color": "rgb(77, 77, 255)",
    "height": 123.33333333333333,
    "key": "1.revenue.block",
    "selector": Array [
      1,
      "revenue",
    ],
    "value": 21,
    "width": 160,
    "x": 170,
    "y": 10,
  },
  Object {
    "color": "rgb(0, 0, 255)",
    "height": 123.33333333333333,
    "key": "2.revenue.block",
    "selector": Array [
      2,
      "revenue",
    ],
    "value": 30,
    "width": 160,
    "x": 330,
    "y": 10,
  },
]
`;

exports[`<Heatmap /> - utils generateVerticalBlocks() should calculate blocks and apply colors 1`] = `
Array [
  Object {
    "color": "rgb(255, 0, 0)",
    "height": 123.33333333333333,
    "key": "0.sale.block",
    "selector": Array [
      0,
      "sale",
    ],
    "value": -3,
    "width": 160,
    "x": 10,
    "y": 256.66666666666663,
  },
  Object {
    "color": "rgb(153, 153, 255)",
    "height": 123.33333333333333,
    "key": "1.sale.block",
    "selector": Array [
      1,
      "sale",
    ],
    "value": 12,
    "width": 160,
    "x": 10,
    "y": 133.33333333333331,
  },
  Object {
    "color": "rgb(255, 0, 0)",
    "height": 123.33333333333333,
    "key": "2.sale.block",
    "selector": Array [
      2,
      "sale",
    ],
    "value": -3,
    "width": 160,
    "x": 10,
    "y": 10,
  },
  Object {
    "color": "rgb(162, 162, 255)",
    "height": 123.33333333333333,
    "key": "0.buy.block",
    "selector": Array [
      0,
      "buy",
    ],
    "value": 11,
    "width": 160,
    "x": 170,
    "y": 256.66666666666663,
  },
  Object {
    "color": "rgb(230, 230, 255)",
    "height": 123.33333333333333,
    "key": "1.buy.block",
    "selector": Array [
      1,
      "buy",
    ],
    "value": 3,
    "width": 160,
    "x": 170,
    "y": 133.33333333333331,
  },
  Object {
    "color": "rgb(162, 162, 255)",
    "height": 123.33333333333333,
    "key": "2.buy.block",
    "selector": Array [
      2,
      "buy",
    ],
    "value": 11,
    "width": 160,
    "x": 170,
    "y": 10,
  },
  Object {
    "color": "rgb(0, 0, 255)",
    "height": 123.33333333333333,
    "key": "0.revenue.block",
    "selector": Array [
      0,
      "revenue",
    ],
    "value": 30,
    "width": 160,
    "x": 330,
    "y": 256.66666666666663,
  },
  Object {
    "color": "rgb(77, 77, 255)",
    "height": 123.33333333333333,
    "key": "1.revenue.block",
    "selector": Array [
      1,
      "revenue",
    ],
    "value": 21,
    "width": 160,
    "x": 330,
    "y": 133.33333333333331,
  },
  Object {
    "color": "rgb(0, 0, 255)",
    "height": 123.33333333333333,
    "key": "2.revenue.block",
    "selector": Array [
      2,
      "revenue",
    ],
    "value": 30,
    "width": 160,
    "x": 330,
    "y": 10,
  },
]
`;
